### ISSUE TITLE
Javadocs for project

### ISSUE THREAD

**Issue by ilrbats** [nus-cs2030/2021-s2#356]
Is it necessary that we write javadocs for our project? Or is it enough to just pass all the test cases for the project since the javadocs checkstyle in the project has been switched off.

**Comment by gableejh** [comment]
not necessary, just pass the test cases will do

**Comment by luthertan98** [comment]
I think it will be good to include javadocs for your project as it is to allow the grader to understand the implementation of your code easily as you can imagine how many porjects they have to grade and, i dont think they have the time to look through every code and understand it. So having the javadocs would help them in understanding your code and implementation

**Comment by JoelToh404** [comment]
I dont think it is necessary to include javadocs for the project

**Comment by evewzhiru** [comment]
i included javadocs to pass checkstyle but if its not in the test cases, i dont think its necessary

**Comment by seancxy** [comment]
The grader in codecrunch doesn't check for this issue, its fine if you don't include them. But if you wanna include them, you can refer to https://www.baeldung.com/javadoc

**Comment by keithcharlesc** [comment]
it's not necessary, it will still pass in codecrunch despite appearing in the local checkstyle checks on your terminal

**Comment by downmeanhigh** [comment]
hi, it's not needed but u should include it to understand your code better

**Comment by SrinivasG99** [comment]
Codecrunch will not reject your code, but checkstyle will still issue a warning

**Comment by Hanyu0118** [comment]
Yes! We do not need to write javadocs for our project. Codecrunch does not test it!

**Comment by vishandi** [comment]
I think while CodeCrunch does not check for javadoc, it is quite helpful for the tutors to check our code when they try to understand it. My tutor told us to put javadocs in every public method.

**Comment by elizabethh07** [comment]
Codecrunch does not require it but it would be good so others can better understand our codes!

**Comment by Jjjetplane** [comment]
Codecrunch does not require javadocs to pass all test cases but would be good to include it so that the tutor grading it can easily understand your code!

**Comment by MatthiasTanWeiSheng** [comment]
Codecrunch does not require you to include javadocs but including it will help your tutors who are grading your project understand your code better. And if it is easier for them to understand, it will be better for you also

**Comment by amaslua** [comment]
it's not required, but not sure if the graders will penalise you even though you get A in codecrunch
